core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -2544,6 +2546,8 @@ private void sendCursor0(
             prepareCommandComplete(true);
         } else {
             prepareSuspended();
+            // Prevents to re-send current record row when buffer is sent fully.
+            resumeProcessor = null;
```

Comment:
`PGJobContextTest` still passes when I comment this line out. May it be the case that the changes in `testCursorFetch` are not enough to reproduce the bug?

Reply:
It fails on mac definitely, perhaps can fail for you if you run it in a loop

Reply:
I tried running it in a loop, but it passes reliably. Probably this behavior has something to do with epoll vs. kqueue specifics. Marking this one as resolved then.